### PR TITLE
mods are working but with limited features

### PR DESCRIPTION
**Pull request by amazingalek**

### CODE REVIEW

File: src/OWML.Launcher/OWML.Manifest.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
 	"author": "Alek",
 	"name": "OWML",
 	"uniqueName": "Alek.OWML",
-	"version": "1.1.9",
+	"version": "2.0.0",
 	"description": "The mod loader and mod framework for Outer Wilds",
 	"minGameVersion": "1.0.7.0",
```
Comment (misternebula) [inline src/OWML.Launcher/OWML.Manifest.json:7]: update game versions (1.1.8 to 1.1.10)

File: src/OWML.ModHelper.Input/ModCommandListener.cs [hunk 2]
```diff
@@ -5,36 +5,36 @@
...
-			}
-		}
-	}	
+        //	if (OWInput.IsNewlyReleased(command))
+        //	{
+        //		OnNewlyReleased?.Invoke(command);
+        //	}
+        //}
+        //}
+    }
```
Comment (misternebula) [inline src/OWML.ModHelper.Input/ModCommandListener.cs:39]: tabs

File: src/OWML.ModHelper.Menus/ModPopupInput.cs [hunk 3]
```diff
@@ -15,12 +15,12 @@ public abstract class ModPopupInput<T> : ModInput<T>
...
-			InputLibrary.menuConfirm,
-			InputLibrary.enter,
-			InputLibrary.enter2
-		};
+		//private readonly List<SingleAxisCommand> _openCommands = new()
+		//{
+		//	InputLibrary.menuConfirm,
+		//	InputLibrary.enter,
+		//	InputLibrary.enter2
+		//};
```
Comment (misternebula) [inline src/OWML.ModHelper.Menus/ModPopupInput.cs:23]: Hm... maybe add some kind of attribute/comment to everything we comment out so we don't miss anything?

File: src/OWML.ModHelper.Menus/ModSelectorInput.cs [hunk 4]
```diff
@@ -7,62 +7,62 @@
...
+        }
 
-		public int SelectedIndex
-		{
-			get => _element.GetCurrentIndex();
-			set => _element.Initialize((value % _count + _count) % _count);
-		}
+        public int SelectedIndex
+        {
+            get => 0;//_element.GetCurrentIndex();
```
Comment (misternebula) [inline src/OWML.ModHelper.Menus/ModSelectorInput.cs]: tabs

File: src/OWML.ModHelper.Input/ModCommandListener.cs (hunk 2 above)
Comment (amazingalek) [reply src/OWML.ModHelper.Input/ModCommandListener.cs:39]: fixed

File: src/OWML.ModLoader/ModLoader.cs [hunk 5]
```diff
@@ -1,69 +1,81 @@
...
+            var goHelper = new GameObjectHelper();
+            var container = CreateContainer(appHelper, goHelper);
 
-			if (owmlConfig == null || owmlManifest == null)
-			{
-				throw new UnityException("Can't load OWML config or manifest.");
-			}
+            var owo = container.Resolve<Owo>();
+            owo.LoadMods();
+        }
```
Comment (misternebula) [inline src/OWML.ModLoader/ModLoader.cs]: tabs

File: src/OWML.ModHelper.Menus/ModPopupInput.cs (hunk 3 above)
Comment (amazingalek) [reply src/OWML.ModHelper.Menus/ModPopupInput.cs:23]: we can search for "//" ;)

File: src/OWML.Patcher/OWPatcher.cs [hunk 6]
```diff
@@ -67,7 +75,31 @@ private void CopyFiles()
...
+				}
+				catch
+				{
+					uncopiedFilesToRoot.Add(filename);
+				}
+			}
+
+			if (uncopiedFilesToRoot.Any())
+			{
+				_writer.WriteLine("Warning - Failed to copy the following files to root :", MessageType.Warning);
```
Comment (misternebula) [inline src/OWML.Patcher/OWPatcher.cs:102]: not needed anymore? we're not using doorstop anymore right?

File: src/OWML.Launcher/OWML.Manifest.json (hunk 1 above)
Comment (amazingalek) [reply src/OWML.Launcher/OWML.Manifest.json]: done

File: src/OWML.Temp/Class1.cs [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+namespace OWML.Temp
+{
+    public class Class1
+    {
+        public void Lol()
+        {
+            var lol = JsonConvert.SerializeObject(new { });
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (amazingalek) [reply src/OWML.Temp/Class1.cs:13]: long story.
short answer: it's needed for now

File: src/OWML.Utils/Container.cs [hunk 8]
```diff
@@ -1,30 +1,31 @@
-﻿using Microsoft.Practices.Unity;
+﻿using Autofac;
```
Comment (misternebula) [inline src/OWML.Utils/Container.cs:1]: there's a nuget package "unity" which would let us not have to rewrite all of this? or did we already try that

File: src/OWML.Utils/JsonHelper.cs [hunk 9]
```diff
@@ -17,14 +17,14 @@ public static T LoadJsonObject<T>(string path)
...
-				return default;
-			}
+			//try
+			//{
+			return JsonConvert.DeserializeObject<T>(json);
+			//}
+			//catch (Exception)
+			//{
+			//	return default;
+			//}
```
Comment (misternebula) [inline src/OWML.Utils/JsonHelper.cs:27]: :/

File: src/OWML.Temp/Class1.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (amazingalek) [reply src/OWML.Temp/Class1.cs:13]: long answer: we need Newtonsoft.Json from Nuget in the Launcher. 
when adding it directly to Launcher it's eaten by the game's version.
adding it to new project, then copying it.
it's only copied to output if it's used, thereby the shitty code

File: src/OWML.Utils/JsonHelper.cs (hunk 9 above)
Comment (amazingalek) [reply src/OWML.Utils/JsonHelper.cs:27]: this isn't a release candidate :P

Comment (amazingalek) [reply src/OWML.Utils/JsonHelper.cs:27]: there's a thousand comments we need to fix, wanna merge this before we get lots of conflicts when working in parallell ;)

File: src/OWML.Utils/Container.cs (hunk 8 above)
Comment (amazingalek) [reply src/OWML.Utils/Container.cs:1]: hmmm? dependency injection is rewritten and works, it's done :)

File: src/OWML.Patcher/OWPatcher.cs (hunk 6 above)
Comment (amazingalek) [reply src/OWML.Patcher/OWPatcher.cs]: fixed

File: src/OWML.Utils/JsonHelper.cs (hunk 9 above)
Comment (amazingalek) [reply src/OWML.Utils/JsonHelper.cs]: fixed this one ;)

**Review by misternebula — APPROVED** [review]

**Pull request merged**